### ISSUE TITLE
Report error

### ISSUE THREAD

**Issue by pablomillaquen** [FMCorz/moodle-block_xp#129]
Hello, I need help with a plugin error. When I select the Report option, the system goes on hold for 60 seconds and returns "Error 500". 

Plugins instalados:
- block_xp v.3.9.0 (2020043003)
- enrol_xp v.1.0.1 (2018051800)
- availability_xp v.2.0.4 (2018051800)
- local_xp v.1.9.0 (Multi) (2020043000)

Site: https://elearningbrasil.sonda.com/

Error log presents the following information:

```
[27-Feb-2022 23:32:46 America/Santiago] PHP Warning:  unlink(/var/moodle-3.8/mod/certificate/pix/borders/Fancy1-blue.jpg): Permission denied in /var/moodle-3.8/lib/tcpdf/tcpdf.php on line 7801
[27-Feb-2022 23:33:09 America/Santiago] PHP Notice:  The renderer was requested too early in the process.<ul style="text-align: left" data-rel="backtrace"><li>line 306 of /local/xp/classes/local/container.php: call to debugging()</li><li>line 98 of /local/xp/classes/local/container.php: call to local_xp\local\container->get_renderer()</li><li>line 54 of /blocks/xp/classes/di.php: call to local_xp\local\container->get()</li><li>line 52 of /local/xp/classes/local/rule/course.php: call to block_xp\di::get()</li><li>line 52 of /blocks/xp/classes/rule.php: call to local_xp\local\rule\course->__construct()</li><li>line 146 of /blocks/xp/classes/ruleset.php: call to block_xp_rule::create()</li><li>line 54 of /blocks/xp/classes/rule.php: call to block_xp_ruleset->import()</li><li>line 231 of /blocks/xp/classes/filter.php: call to block_xp_rule::create()</li><li>line 248 of /blocks/xp/classes/filter.php: call to block_xp_filter->load_rule()</li><li>line 65 of /local/xp/classes/local/rule/filters_calculator.php: call to block_xp_filter->match()</li><li>line 80 of /local/ in /var/moodle-3.8/lib/weblib.php on line 3241
[27-Feb-2022 23:35:16 America/Santiago] PHP Notice:  The renderer was requested too early in the process.<ul style="text-align: left" data-rel="backtrace"><li>line 306 of /local/xp/classes/local/container.php: call to debugging()</li><li>line 98 of /local/xp/classes/local/container.php: call to local_xp\local\container->get_renderer()</li><li>line 54 of /blocks/xp/classes/di.php: call to local_xp\local\container->get()</li><li>line 52 of /local/xp/classes/local/rule/course.php: call to block_xp\di::get()</li><li>line 52 of /blocks/xp/classes/rule.php: call to local_xp\local\rule\course->__construct()</li><li>line 146 of /blocks/xp/classes/ruleset.php: call to block_xp_rule::create()</li><li>line 54 of /blocks/xp/classes/rule.php: call to block_xp_ruleset->import()</li><li>line 231 of /blocks/xp/classes/filter.php: call to block_xp_rule::create()</li><li>line 248 of /blocks/xp/classes/filter.php: call to block_xp_filter->load_rule()</li><li>line 65 of /local/xp/classes/local/rule/filters_calculator.php: call to block_xp_filter->match()</li><li>line 80 of /local/ in /var/moodle-3.8/lib/weblib.php on line 3241
[27-Feb-2022 23:35:16 America/Santiago] PHP Warning:  unlink(/var/moodle-3.8/mod/certificate/pix/borders/Fancy1-blue.jpg): Permission denied in /var/moodle-3.8/lib/tcpdf/tcpdf.php on line 7801
[27-Feb-2022 23:37:58 America/Santiago] PHP Notice:  The renderer was requested too early in the process.<ul style="text-align: left" data-rel="backtrace"><li>line 306 of /local/xp/classes/local/container.php: call to debugging()</li><li>line 98 of /local/xp/classes/local/container.php: call to local_xp\local\container->get_renderer()</li><li>line 54 of /blocks/xp/classes/di.php: call to local_xp\local\container->get()</li><li>line 52 of /local/xp/classes/local/rule/course.php: call to block_xp\di::get()</li><li>line 52 of /blocks/xp/classes/rule.php: call to local_xp\local\rule\course->__construct()</li><li>line 146 of /blocks/xp/classes/ruleset.php: call to block_xp_rule::create()</li><li>line 54 of /blocks/xp/classes/rule.php: call to block_xp_ruleset->import()</li><li>line 231 of /blocks/xp/classes/filter.php: call to block_xp_rule::create()</li><li>line 248 of /blocks/xp/classes/filter.php: call to block_xp_filter->load_rule()</li><li>line 65 of /local/xp/classes/local/rule/filters_calculator.php: call to block_xp_filter->match()</li><li>line 80 of /local/ in /var/moodle-3.8/lib/weblib.php on line 3241
[27-Feb-2022 23:39:41 America/Santiago] PHP Notice:  The renderer was requested too early in the process.<ul style="text-align: left" data-rel="backtrace"><li>line 306 of /local/xp/classes/local/container.php: call to debugging()</li><li>line 98 of /local/xp/classes/local/container.php: call to local_xp\local\container->get_renderer
```

Thanks for your help

**Comment by FMCorz** [comment]
Hi @pablomillaquen 

Would you please update to the latest versions and confirm whether this is still happening?

Thank you!

**Comment by FMCorz** [comment]
Also please note that those logs do not cause any fatal error, at least not for the one related to Level Up. The one related to mod_certificate might be.

**Comment by FMCorz** [comment]
Closing this. The notices shown were fixed in Level Up Plus v1.10.2 (Jan 2021).

Please comment if the error reported persists.
